clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -244,11 +247,11 @@ public void updateFetchPositions(Set<TopicPartition> partitions) {
                 continue;
 
             if (subscriptions.isOffsetResetNeeded(tp)) {
-                resetOffset(tp);
+                resetOffsets(Collections.singleton(tp));
```

Comment:
The ordering is a little tricky when setting the offsets. The logic is like this:

1. Fetch offsets which have been explicitly requested by the user through `seekToBeginning` or `seekToEnd`. (We do this first to avoid unnecessary dependence on the group coordinator for consumers using manual assignment.)
2. Fetch committed offsets for any positions which haven't been updated.
3. Now reset offsets for any partitions which are left (i.e. those that were not explicitly seeked on and which have no committed offset).

So I think we still need to handle this here. My question for this patch is which case are you trying to optimize for? The case where you have done a `seek` on all the partitions?